### PR TITLE
Rehp simplification #2

### PR DESCRIPTION
**Pull request by Lupus**

### CODE REVIEW

**Comment by robz** [comment]
@Lupus all of this is great! But I'd prefer it if we could keep += in the output, since I think it makes it look cleaner. @jordwalke suggested moving that simplification into javascript_from_rehp and php_from_rehp. What do you think?

**Comment by Lupus** [comment]
You mean sugar `a = a + 1` back into `a += 1` in `javascript_from_rehp` and `php_from_rehp`? Last time @jordwalke said that `a = a + 1` is perfectly debuggable, that is - user opening up the compiled code in JS debugger will be able to read it just fine, so I left it as is despite the fact that it's a small regression. Do you suggest it's worth fixing within this PR?

**Comment by robz** [comment]
I don't feel very strongly about it, just thought it would be nice to have parity with the output before. We can merge this--I'll just make an issue to track simplifying to += later

**Comment by Lupus** [comment]
> We can merge this--I'll just make an issue to track simplifying to += later

Makes sense to me, otherwise this PR might get stale. I don't have time right now to take a look into sugaring it in JS/Hack, yet it does not really block these backends and is indeed a stylistic improvement that can be implemented at any point later.

**Comment by Lupus** [comment]
@robz I've had discussion regarding this with @jordwalke and we agreed to postpone `++/+=` change until you're ready with reverse transformations in backends. This way we can merge this PR right away if there are no further suggestions (although quick glance over the diff is appreciated, rebase was not completely trivial).

File: compiler/lib/javascript_from_rehp.re [hunk 1]
```diff
@@ -182,16 +182,10 @@ and from_statement = e =>
...
-      | None => None
-      | Some((ident, block)) => Some((ident, from_statement_list(block)))
-      };
-
-    let block_opt = Stdlib.Option.map(~f=from_statement_list, block_opt);
-    Try_statement(block, ident_block_opt, block_opt);
+  | Rehp.Try_statement(b1, (ident, b2)) =>
+    let b1 = from_statement_list(b1);
+    let ident_block = (ident, from_statement_list(b2));
+    Try_statement(b1, Some(ident_block), None);
```
Comment (robz) [inline compiler/lib/javascript_from_rehp.re:201]: I suppose we could also change the JS and PHP ASTs

**Review by robz — APPROVED** [review]
Looks good to me!

File: compiler/lib/javascript_from_rehp.re (hunk 1 above)
Comment (Lupus) [reply compiler/lib/javascript_from_rehp.re:201]: Yeah, but that could be done any time later.

**Pull request merged**